File: webf/lib/src/svg/text.dart
/*
 * Copyright (C) 2022-present The WebF authors. All rights reserved.
 */

import 'package:webf/rendering.dart';
import 'package:webf/svg.dart';


class SVGTextElement extends SVGTextPositioningElement {
  @override
  get presentationAttributeConfigs => super.presentationAttributeConfigs..addAll([]);

  SVGTextElement(super.context) {}

  @override
  RenderBoxModel createRenderSVG({RenderBoxModel? previous, bool isRepaintBoundary = false}) {
    return RenderSVGText(renderStyle: renderStyle);
  }
}

File: bridge/core/rust_api/text.cc
/*
 * Copyright (C) 2022-present The WebF authors. All rights reserved.
 */

#include "text.h"
#include "core/dom/character_data.h"

namespace webf {

TextNodeRustMethods::TextNodeRustMethods(CharacterDataRustMethods* super_rust_method)
    : character_data(super_rust_method) {}

}  // namespace webf